app/src/main/java/org/schabi/newpipe/util/ReleaseUtil.kt

```diff
@@ -0,0 +1,160 @@
+
+object ReleaseUtil {
+    // Public key of the certificate that is used in NewPipe release versions
+    private const val RELEASE_CERT_PUBLIC_KEY_SHA1 =
+        "B0:2E:90:7C:1C:D6:FC:57:C3:35:F0:88:D0:8F:50:5F:94:E4:D2:15"
+
+    @JvmStatic
+    fun isReleaseApk(): Boolean {
+        return certificateSHA1Fingerprint == RELEASE_CERT_PUBLIC_KEY_SHA1
+    }
+
+    /**
+     * Method to get the APK's SHA1 key. See https://stackoverflow.com/questions/9293019/#22506133.
+     *
+     * @return String with the APK's SHA1 fingerprint in hexadecimal
+     */
+    private val certificateSHA1Fingerprint: String
+        get() {
+            val app = App.getApp()
+            val signatures: List<Signature> = try {
+                PackageInfoCompat.getSignatures(app.packageManager, app.packageName)
+            } catch (e: PackageManager.NameNotFoundException) {
+                showRequestError(app, e, "Could not find package info")
+                return ""
+            }
+            if (signatures.isEmpty()) {
+                return ""
+            }
+            val x509cert = try {
+                val cert = signatures[0].toByteArray()
+                val input: InputStream = ByteArrayInputStream(cert)
+                val cf = CertificateFactory.getInstance("X509")
+                cf.generateCertificate(input) as X509Certificate
+            } catch (e: CertificateException) {
+                showRequestError(app, e, "Certificate error")
+                return ""
+            }
+
+            return try {
+                val md = MessageDigest.getInstance("SHA1")
+                val publicKey = md.digest(x509cert.encoded)
+                byte2HexFormatted(publicKey)
+            } catch (e: NoSuchAlgorithmException) {
+                showRequestError(app, e, "Could not retrieve SHA1 key")
+                ""
+            } catch (e: CertificateEncodingException) {
+                showRequestError(app, e, "Could not retrieve SHA1 key")
+                ""
+            }
+        }
+
+    private fun byte2HexFormatted(arr: ByteArray): String {
+        val str = StringBuilder(arr.size * 2)
+        for (i in arr.indices) {
+            var h = Integer.toHexString(arr[i].toInt())
+            val l = h.length
+            if (l == 1) {
+                h = "0$h"
+            }
+            if (l > 2) {
+                h = h.substring(l - 2, l)
+            }
+            str.append(h.uppercase())
+            if (i < arr.size - 1) {
+                str.append(':')
+            }
+        }
+        return str.toString()
+    }
+
+    /**
+     * Method to compare the current and latest available app version.
+     * If a newer version is available, we show the update notification.
+     *
+     * @param versionName    Name of new version
+     * @param apkLocationUrl Url with the new apk
+     * @param versionCode    Code of new version
+     */
+    fun compareAppVersionAndShowNotification(
```

Comment:
Why is this function here? I'd move it to `NewVersionWorker` as it is used only there and is strictly related to the worker

Reply:
I felt that it was unnecessarily cluttering up the `NewVersionWorker`. I can move it back, I suppose.

Reply:
In my opinion yes, it is not a "release utility" that could be useful throughout the app, it is a method that creates the notification as an output to the worker

Reply:
Okies, done 👍